Do not reuse a connection that fail while uploading (#863)
Azure/azure-sdk-for-cpp: sdk/core/azure-core/CHANGELOG.md

### Bug Fixes

- Prevent pipeline of length zero to be created.

> - Avoid re-using a connection when an uploading request fail when using CurlTransport.

Reviewer: ```suggestion
- Avoid re-using a connection when a request to upload data fails while using the `CurlTransport`.
```
